src/dmd/root/region.d

```diff
@@ -86,9 +89,28 @@ struct Region
      */
     bool contains(void* p)
     {
-        foreach (h; array[0 .. used])
+        if (!used)
+            return false;
+        if (!isSorted)
+            sort();
```

Comment:
This looks dangerous: if new chunks were allocated and released without `contains` being called, they might get moved to a position before `used` while actually being unused. Inserting them during allocation does not have that problem and is slightly more efficient as it is executed there very seldom.

Reply:
The regions are not set up to be a stack. That would be a separate PR.

Reply:
This is not related to being a stack. Consider 3 chunks allocated but unsorted, and 2 chunks used until `contains` is actually called: the unused chunk might get sorted to the front, while a used one will be at the end and will be reused in a later allocation.